### ISSUE TITLE
Notification Level for Check In only

### ISSUE THREAD

**Issue by spacezorro** [jdholtz/auto-southwest-check-in#280]
### Description

I want to send all notifications to Email but have successful/failed checkins only go to SMS

Where 1 is everything, 2 is successful checkins/errors, and 3 is errors only
I want to do
{
  "notification_level": 1,
  "notification_urls": "mailtos://ALLthestuffgoeshere",
    "accounts": [
      {
        "username": "1234567", "password": "hunter2",  "notification_level": 2,
        "notification_urls": "sns://gobbledegook/us-west-1/+15555551212"
      },
      {
        "username": "7654321", "password": "hunter2",  "notification_level": 2,
        "notification_urls": "sns://gobbledegook/us-west-1/+15555552222"
      }

etc...

That way it keeps the SMS to a minimum instead of SMS bombing someone when the server/docker restarts

**Comment by jdholtz** [comment]
Yes, this will be a good feature to add. I’ll add it in the same release as the notifications I explained in #249 which will be by next release.

**Comment by davidkassa** [comment]
If you're considering updating notifications, can you make the notification_level and url paired? For example:
{
"username": "7654321", "password": "hunter2",
    "notifications": [
        { "notification_level": 1, "notification_urls": "sns://gobbledegook/us-west-1/+15555552222"}, // admin number / notification
        { "notification_level": 4, "notification_urls": "sns://gobbledegook/us-west-1/+15555552222"} // family number / notification
    ]
}

**Comment by jdholtz** [comment]
> If you're considering updating notifications, can you make the notification_level and url paired?

I partly misunderstood the original feature request. I believe this is also what @spacezorro is requesting (there are 2 feature requests). I've reverted the changes done in #281 relating to this feature so this could be discussed further.

With the configuration provided by spacezorro, the account notification level would actually override the global level, meaning even the email notifications would be sent only error messages.

Here's what I'm thinking for how this will be specified for the config file. I don't want it to be too complicated either, so let me know what you guys think.

```json
{
    "notifications": [{"url": "smtp://email"}],
    "accounts": [
        {"username": "...", "password": "...", "notifications": [
            {"url": "sms://text", "level": 2}, 
            {"url": "discord://channel"}
        ]},
        {"username": "...", "password": "..."}
    ]
}
```

For account 1, there would be notifications sent to three places (the levels will be based off of v7.5, not #281): 
- Email with level 1, SMS with level 2, and Discord with level 1 

For account 2:
- Email with level 1

Any level not specified will default to the current default notification level. If the same URL exists in both the account-specific and global notifications, the account-specific level will override the global level.

**Comment by spacezorro** [comment]
That's perfect. I run the script for family and my wife/kids only want checkin messages. They don't care about the scheduling or fare checking... But I do so your most recent example is exactly what I was going for. 

Does there need to be more levels for "checkin success/fail only"?

**Comment by jdholtz** [comment]
> Does there need to be more levels for "checkin success/fail only"?

Yes, I’ll add that as well. I’ll make a separate PR to handle both this and the new config format.

**Comment by jdholtz** [comment]
Both the notification check-in level and new notification configuration was added in #328. Would you guys be able to test it out and let me know how it works for you and if you have any issues? The new configuration should reflect my comment above, but documentation can be found for it in the CONFIGURATION.md doc (also let me know if this documentation should be improved). Thanks!

**Comment by jdholtz** [comment]
Hey @spacezorro and @davidkassa, would you be able to look over the config changes to the notification system in #328 and provide any feedback on how well it achieves what you are requesting in this issue?

**Comment by spacezorro** [comment]
Sorry I commented in #328 but didn't comment here.

This is great. My wife only cares about "Did it checkin or not" all the other "is it working" is up to me :)

What I was going for I can do now with..
Her cell phone Level3, email at Level2, and my cell phone at Level4

She gets the check fare to her email, Checkin to her phone, and if it blows up I get a message to fix it :)

**Comment by jdholtz** [comment]
Thanks @spacezorro and @davidkassa for your suggestions for improvement. The new notification system will be in the next version (and is currently in the develop branch).
